### PR TITLE
Story app: Create and add story app to project

### PR DESCRIPTION
**Pull request by Yoavsc129**
Changed app name from mystory to story

### CODE REVIEW

**Review by ShellyGolden — APPROVED** [review]

**Review by LiorKGOW — APPROVED** [review]
LGTM good job overall @Yoavsc129

**Comment by adiabramovitch** [comment]
@Yoavsc129 you have conflicts to resolve

File: story/models.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+from django.db import models
+from django.utils import timezone
+from house.models import House
+
+
+class Story(models.Model):
+    house = models.ForeignKey(House, on_delete=models.CASCADE)
+    story_date = models.DateTimeField(default=timezone.now)
+    story_text = models.TextField()
+
```
Comment (adiabramovitch) [inline story/models.py:10]: I would change the variable name to align with the ERD you have made. 
for example, date is supposed to be the date the story was last edited. 

what about CRUD functions? for example: https://github.com/redhat-beyond/Xpense/blob/e00f90465ec146a84a3acd052120c644001f83b6/house/models.py#L60

File: story/tests.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+city = City(name=CITY, country=country)
+house = House(name=HOUSE_NAME, public=HOUSE_PUBLIC, country=country, city=city, income=1, children=1)
+
+
+@pytest.mark.django_db()
+def test_new_item():
+    country.save()
+    city.save()
+    house.save()
+    item = Story(story_date=TEMP_DATE, story_text=TEMP_STORY, house=house)
```
Comment (adiabramovitch) [inline story/tests.py:25]: ditto

File: story/tests.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+CITY = "South Park"
+HOUSE_PARENT_PROFESSION_1 = "Teacher"
+HOUSE_PARENT_PROFESSION_2 = "Student"
+country = Country(name=COUNTRY)
+city = City(name=CITY, country=country)
+house = House(name=HOUSE_NAME, public=HOUSE_PUBLIC, country=country, city=city, income=1, children=1)
+
+
+@pytest.mark.django_db()
+def test_new_item():
```
Comment (adiabramovitch) [inline story/tests.py:21]: What is the item? it is supposed to be more informative.

File: story/tests.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    item.save()
+
+    assert item.story_text == TEMP_STORY
+    assert item.story_date == TEMP_DATE
+    assert item.house.house_id == house.house_id
+
+
+@pytest.fixture
+def story0():
+    return Story(story_date=TEMP_DATE, story_text=TEMP_STORY, house=house)
```
Comment (adiabramovitch) [inline story/tests.py:35]: you should be persistent throughout the project. 
You have decided as far as I know to use CRUD methods (https://github.com/redhat-beyond/Xpense/blob/e00f90465ec146a84a3acd052120c644001f83b6/house/models.py#L60)

File: story/tests.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    def test_persist_story(self, story0):
+        country.save()
+        city.save()
+        house.save()
+        story0.save()
+
+        assert story0 in Story.objects.all()
+
+    @pytest.mark.django_db()
+    def test_deletecascade_house(self, story0):
```
Comment (adiabramovitch) [inline story/tests.py:53]: if you chose that name I would change it to: 
```
test_delete_cascade_house
```
IMHO, I would change it to a more informative one like: 
```
test_story_is_deleted_after_house_was_deleted
```

**Review by adiabramovitch — CHANGES_REQUESTED** [review]

File: story/tests.py (hunk 5 above)
Comment (Yoavsc129) [reply story/tests.py]: I adreesed your comments @adiabramovitch, please take another look

**Comment by Yoavsc129** [comment]
Because of the on_delete-cascade in the house foreign key

File: house/migrations/0004_alter_house_children_alter_house_house_id_and_more.py [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+            model_name='house',
+            name='house_id',
+            field=models.UUIDField(default=uuid.uuid4, editable=False, primary_key=True, serialize=False),
+        ),
+        migrations.AlterField(
+            model_name='house',
+            name='income',
+            field=models.IntegerField(blank=True, null=True, validators=[django.core.validators.MinValueValidator(0)]),
+        ),
+    ]
```
Comment (ronyyosef) [inline house/migrations/0004_alter_house_children_alter_house_house_id_and_more.py:30]: What is that code doing? do we need that?

Comment (Yoavsc129) [reply house/migrations/0004_alter_house_children_alter_house_house_id_and_more.py:30]: We do. This file was created as part of the migrations and is required for the delete-cascade functionality

**Review by adiabramovitch — APPROVED** [review]

**Review by ShellyGolden — APPROVED** [review]

**Comment by ronyyosef** [comment]
This PR needs to be close as we talk in our today's meeting

**Pull request closed without merging**